distro/config.go

```diff
@@ -62,6 +86,81 @@ func (c config) Validate() error {
 	return nil
 }
 
+type nonePropagatorType struct{ propagation.TextMapPropagator }
+
+// nonePropagator signals the disablement of setting a global
+// TextMapPropagator.
+var nonePropagator = nonePropagatorType{}
```

Comment:
nit: maybe just use `nil` instead? having `nonePropagatorType` gives a false impression that the interface is implemented. It would solve "my problems" that `nonePropagator` is not a "NullObject" having a no-op implementation. `nil` is nothing - very obvious 😄 feel free to resolve if you disagree or do not want to lose time on it

Reply:
Yeah, that's a good idea, and one I looked into. The issue that it runs into is that when the user passes `nil` to the `WithPropagator` meaning setting a propagator should be disabled and at the same time `OTEL_PROPAGATOR` is defined as non-`"none"`. Without a sentinel value to signal this disablement from the first option it is indistinguishable from not setting that option.

Reply:
I see. Thanks. Can you add this info as a comment? 😉

Reply:
Probably as an alternative you can filter out the `nil` before passing to `WithPropagator`.

Reply:
@pellared I'm definitely open to suggestions if you have them on how to resolve this issue and still use the `nil` value. I agree it would clean things up nicely.

Reply:
> Probably as an alternative you can filter out the `nil` before passing to `WithPropagator`.

I'm not sure I follow. Do you mean we could change the behavior and drop any passed nil?

> Can you add this info as a comment?

I added [this](https://github.com/signalfx/splunk-otel-go/pull/295/files#diff-8d2c05f32348fdabc91f4819ba3cf985b618d684a8c79c1111b7c8794164be92R220-R222). Did you have a suggestion for more?

Reply:
That is fine. Thanks a lot 👍